defopt.py

```diff
@@ -118,8 +118,23 @@ def _create_parser(funcs, *args, **kwargs):
     return parser
 
 
-class _Formatter(RawTextHelpFormatter, ArgumentDefaultsHelpFormatter):
-    pass
+class _Formatter(RawTextHelpFormatter):
+
+    # modified from ArgumentDefaultsHelpFormatter to add type information, and
+    # remove defaults for varargs (`ZERO_OR_MORE`).
+    def _get_help_string(self, action):
+        info = []
+        if action.type is not None and '%(type)' not in action.help:
+            info.append('%(type)s')
+        if (action.default != argparse.SUPPRESS
+                and '%(default)' not in action.help
+                and action.option_strings
+                and action.nargs != argparse.ZERO_OR_MORE):
```

Comment:
Well, the empty list does appear a bit redundant in
```
usage: foo.py [-h] [-a [ARGS [ARGS ...]]]

optional arguments:
  -h, --help            show this help message and exit
  -a [ARGS [ARGS ...]], --args [ARGS [ARGS ...]]
                        Some help (default: [])
```

I *could* additionally check whether the default is empty, but I cannot distinguish between
```
def main(*args):
    """
    Parameters
    ----------
    args : List[int]
        Some help
    """
```
and
```
def main(args=[]):
    """
    Parameters
    ----------
    args : List[int]
        Some help
    """
```
which produce exactly the same parser objects (unless we attach additional information at parser construction time).

Which lets me notice an additional issue: napoleon indicates that varargs should be documented using
```
Parameters
----------
*args : ...
```
i.e. including the star, unescaped (and similarly for google-style), but defopt only recognizes the parameter name *without* the star).